### PR TITLE
update futures dependency to 0.3

### PR DESCRIPTION
**Pull request by aav**
update  futures dependency to 0.3

### CODE REVIEW

**Comment by aav** [comment]
@googlebot I signed it!

File: async-coap/src/datagram/allow_udp_socket.rs [hunk 1]
```diff
@@ -91,13 +90,9 @@ impl AllowStdUdpSocket {
...
-            if opt_mut.is_none() {
-                *opt_mut = Some(Delay::new(d));
-                delay = opt_mut.as_mut().unwrap();
-            } else {
-                delay = opt_mut.as_mut().unwrap();
-                delay.reset(d);
-            }
+
+            *opt_mut = Some(Delay::new(d));
+            delay = opt_mut.as_mut().unwrap();
```
Comment (darconeous) [inline async-coap/src/datagram/allow_udp_socket.rs]: This change doesn't immediately look related to the futures update. Can you elaborate?

Comment (aav) [reply async-coap/src/datagram/allow_udp_socket.rs]: hm.. i was almost 100% sure, that Delay::reset has been removed from the API, at some point. I will check and if it is still there, obviously, this change does not make sense, and I will revert it.

Comment (aav) [reply async-coap/src/datagram/allow_udp_socket.rs]: just pushed the commit, which reverses unneeded Delay re-creation. Although, some small change is still needed - Delay::reset chnaged it's signature.

**Comment by aav** [comment]
compilation fails due to regression in 1.42-nightly :(

https://github.com/rust-lang/rust/issues/67743

**Comment by darconeous** [comment]
With this change we can build on stable now, right?

**Comment by aav** [comment]
Yes, it builds on stable

**Pull request merged**